File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.ts
function longestConsecutive(nums: number[]): number {
    let set = new Set(nums)
    let ans = 0

    for (const num of set) {
        let len = 1
        if (!set.has(num-1)) {
            while (set.has(num+len)) {
                len++
            }
            ans = Math.max(ans, len)
        }
    }
    return ans
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        ans = 0
        s = set(nums)

        for num in s:
            count = 1
            if (num-1) not in s:
                while (num+count) in s:
                    count += 1
            ans = max(ans, count)

        return ans
